builders/integrations/bridges/wormhole.md

```diff
@@ -0,0 +1,422 @@
+
+It’s best to tackle the configuration and setup of these four components in order, so start with the spy node. First, in the command line, make sure that you are in the _example-project_ directory. The spy node uses docker, so ensure that docker is active before attempting to start the node. The command to start the container is long, so to simplify things, it’s been added as an npm script to the repository. Just run:
+
+```
+npm run spy
+```
+
+First, you should see a few logs from the startup of the docker container. Then, a lot of logs should be spamming the console. These are all the VAAs that are going through the Wormhole testnet, and there are a lot! Don’t worry, you won’t have to decipher any of these logs: the code can do that for us. Leave this running in the background and get another terminal instance to move on to the next step.
+
+![Run the spy relayer](/images/builders/integrations/bridges/wormhole/wormhole-6.png)
+
+### Setting up the Listener Component {:setting-up-the-listener-component}
+
+Now to break down the custom code and configurable component of the relayer. The listener component, aptly named, listens to the spy node for relevant messages. To define what the relevant messages are, you must edit a config file.
+
+In _example-project/plugins/simplegeneralmessage_plugin/config/devnet.json_, there exists an array named _spyServiceFilters_. Each object within this array whitelists a contract’s VAAs as relevant to the relayer. The object takes in a _chainId _(Wormhole _chainId_) and an _emitterAddress_. For example, in the image below, the first object will watch for VAAs sent by 0x428097dCddCB00Ab65e63AB9bc56Bb48d106ECBE on Moonbase Alpha (Wormhole _chainId_ is 16).
+
+Be sure to edit the _spyServiceFilters_ array so that the relayer listens to the two contracts that you deployed.
+
+
+```javascript
+ "spyServiceFilters": [
+   {
+     "chainId": 16,
+     "emitterAddress": "0x428097dCddCB00Ab65e63AB9bc56Bb48d106ECBE"
+   },
+   {
+     "chainId": 10,
+     "emitterAddress": "0x5017Fd40aeA8Ab94693bE41b3bE4e90F45860bA4"
+   }
+ ]
+```
+
+
+In the _simplegeneralmessage_plugin_ folder, open up _src/plugin.ts_. This file contains plugin code for both the listener and executor components of the relayer, but the comments should make it obvious which functions are relevant to which component. Snippets of the file are shown below and you should be following along, but in case you aren’t, the entire file can be accessed in [its Github repository](https://github.com/jboetticher/relayer-engine/blob/main/example-project/plugins/simplegeneralmessage_plugin/src/plugin.ts){target=_blank}.
```

Comment:
```suggestion
In the _simplegeneralmessage_plugin_ folder, open up _src/plugin.ts_. This file contains plugin code for both the listener and executor components of the relayer, but the comments should make it obvious which functions are relevant to which component. Snippets of the file are shown below and you should be following along, but in case you aren’t, the entire file can be accessed in [its GitHub repository](https://github.com/jboetticher/relayer-engine/blob/main/example-project/plugins/simplegeneralmessage_plugin/src/plugin.ts){target=_blank}.
```

Reply:
check other mentions of github please